src/com/owncloud/android/ui/fragment/OCFileListFragment.java

```diff
@@ -335,60 +343,93 @@ private void removeFabLabels() {
     }
 
     private void registerLongClickListener() {
-        getListView().setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
-            public boolean onItemLongClick(AdapterView<?> arg0, View v,
-                                           int index, long arg3) {
-                showFileAction(index);
-                return true;
-            }
-        });
-    }
+        AbsListView listView = getListView();
+        setChoiceMode(ListView.CHOICE_MODE_MULTIPLE_MODAL);
+        setMultiChoiceModeListener(new AbsListView.MultiChoiceModeListener() {
 
+            private Menu menu;
 
-    private void showFileAction(int fileIndex) {
-        Bundle args = getArguments();
-        PopupMenu pm = new PopupMenu(getActivity(),null);
-        Menu menu = pm.getMenu();
+            @Override
+            public void onItemCheckedStateChanged(ActionMode mode, int position, long id, boolean checked) {
+                final int checkedCount = getListView().getCheckedItemCount();
+                // TODO Tobi extract to values
+                mode.setTitle(checkedCount + " selected");
 
-        boolean allowContextualActions =
-                (args == null) ? true : args.getBoolean(ARG_ALLOW_CONTEXTUAL_ACTIONS, true);
+                if (checked) {
+                    mAdapter.setNewSelection(position, checked);
+                } else {
+                    mAdapter.removeSelection(position);
+                }
 
-        if (allowContextualActions) {
-            MenuInflater inflater = getActivity().getMenuInflater();
+                OCFile targetFile = null;
+                if (checkedCount > 0 && checkedCount <= MIN_FILES_FOR_MULTISELECT) {
+                    targetFile = (checkedCount == MIN_FILES_FOR_MULTISELECT)
+                        ? mFile : mAdapter.getCheckedItems().get(0);
+
+                    if (mContainerActivity.getStorageManager() != null) {
+                        FileMenuFilter mf = new FileMenuFilter(
+                            targetFile,
+                            mContainerActivity.getStorageManager().getAccount(),
+                            mContainerActivity,
+                            getActivity(),
+                            mAdapter.getCheckedItems().size() == MIN_FILES_FOR_MULTISELECT
+                        );
+                        mf.filter(menu);
+                    }
+
+                } else {
+                    getListView().clearChoices();
+                    menu.clear();
+                    menu.close();
```

Comment:
Sorry, I'm checking it right now.

It's from my last changes to avoid the creation of new menus each time you select or deselect an item, and just filter the actions shown in the menu instead.